### ISSUE TITLE
Port request: weak points

### ISSUE THREAD

**Issue by LyleSY** [cataclysmbnteam/Cataclysm-BN#1993]
### Is your feature request related to a problem? Please describe.

The weak points system in DDA is a lot of fun and I think it fits the spirit of BN.

### Describe the solution you'd like

Port weak points, especially https://github.com/CleverRaven/Cataclysm-DDA/pull/51304 , https://github.com/CleverRaven/Cataclysm-DDA/pull/51372 , https://github.com/CleverRaven/Cataclysm-DDA/pull/51770 , and  https://github.com/CleverRaven/Cataclysm-DDA/pull/51822

### Describe alternatives you've considered

Just leave the content out, a system that is even better but keeps the same JSON format

**Comment by scarf005** [comment]
could it also handle NPCs? anyways seems like a good idea

**Comment by LyleSY** [comment]
Yes, NPCs can hit the weakpoints too, but they have their own confusing armor system

**Comment by scarf005** [comment]
yeah, npc armor is kinda weird because iirc they do not feel pain
